comid/tdx-profile/teedigest.go

```diff
@@ -0,0 +1,189 @@
+func NewTeeDigestExpr(op uint, val Digests) (*TeeDigest, error) {
+	if len(val) == 0 {
+		return nil, fmt.Errorf("zero len value for TeeDigestExpr")
+	}
+	switch op {
+	case MEM, NMEM: // Allowed operators are either Members or Non Members
+		set, err := NewTaggedSetDigestExpression(op, val)
+		if err != nil {
+			return nil, fmt.Errorf("zero len value for TeeDigestExpr")
+		}
+		return &TeeDigest{val: *set}, nil
+	default:
+		return nil, fmt.Errorf("invalid operator : %d", op)
+	}
+}
+
+// AddTeeDigest add supplied Digests to existing TeeDigest
+func (o *TeeDigest) AddTeeDigest(op uint, val Digests) (*TeeDigest, error) {
+	if len(val) == 0 {
+		return nil, fmt.Errorf("zero len value for TeeDigestExpr")
+	}
+	switch t := o.val.(type) {
+	case Digests:
+		t = append(t, val...)
+		o.val = t
+	case TaggedSetDigestExpression:
+		if t.SetOperator != Operator(op) {
+			return nil, fmt.Errorf("operator mis-match TeeDigest Op: %d, Input Op: %d", t.SetOperator, op)
+		}
+		for _, value := range val {
+			t.SetDigest = append(t.SetDigest, value)
+		}
+		o.val = t
+	}
+	return o, nil
+}
+
+// Valid checks for validity of TeeDigest and
+// returns an error, if invalid
+func (o TeeDigest) Valid() error {
+	if o.val == nil {
+		return fmt.Errorf("TeeDigest not set")
+	}
+	switch t := o.val.(type) {
+	case Digests:
+		if len(t) == 0 {
+			return fmt.Errorf("TeeDigest not set")
+		}
+	case TaggedSetDigestExpression:
+		if t.SetOperator != MEM && t.SetOperator != NMEM {
+			return fmt.Errorf("invalid operator in a TeeDigest: %d", t.SetOperator)
+		}
+		if len(t.SetDigest) == 0 {
+			return fmt.Errorf("TeeDigest not set")
+		}
+	}
+	return nil
+}
+
+// IsDigests returns true if TeeDigest is TeeDigest
+func (o TeeDigest) IsDigests() bool {
+	return isType[Digests](o.val)
+}
+
+// IsDigestExpr returns true if TeeDigest is DigestExpr
+func (o TeeDigest) IsDigestExpr() bool {
+	return isType[TaggedSetDigestExpression](o.val)
+}
+
+// GetDigestExpr returns a Digest Expression
+func (o TeeDigest) GetDigestExpr() (*TaggedSetDigestExpression, error) {
+	if err := o.Valid(); err != nil {
+		return nil, fmt.Errorf("invalid TEEDigest: %w", err)
+	}
+
+	switch t := o.val.(type) {
+	case TaggedSetDigestExpression:
+		return &t, nil
+	default:
+		return nil, fmt.Errorf("invalid type: %T", t)
+	}
+}
+
+func (o TeeDigest) GetDigest() (Digests, error) {
+	switch t := o.val.(type) {
+	case Digests:
+		return t, nil
+	default:
+		return nil, fmt.Errorf("invalid type: %T", t)
+	}
+}
+
+func (o TeeDigest) MarshalJSON() ([]byte, error) {
+	var (
+		v   encoding.TypeAndValue
+		b   []byte
+		err error
+	)
+	switch t := o.val.(type) {
+	case []string:
+		b, err = json.Marshal(t)
+		if err != nil {
+			return nil, err
+		}
+		v = encoding.TypeAndValue{Type: DigestType, Value: b}
+	case TaggedSetDigestExpression:
+		b, err = json.Marshal(t)
+		if err != nil {
+			return nil, err
+		}
+		v = encoding.TypeAndValue{Type: DigestExprType, Value: b}
+	default:
+		fmt.Printf("YOGESH CAME HERE")
```

Comment:
LOL

```suggestion
		fmt.Printf(“Bart Simpsons was HERE")
```

Reply:
Well spotted, was debugging a strange issue. Will remove is ASAP!